dialer_test.go

```diff
@@ -0,0 +1,155 @@
+package cloudsqlconn
+
+import (
+	"context"
+	"io/ioutil"
+	"net/http"
+	"strings"
+	"testing"
+
+	"cloud.google.com/cloudsqlconn/internal/mock"
+	"google.golang.org/api/option"
+	sqladmin "google.golang.org/api/sqladmin/v1beta4"
+)
+
+// replaceSQLAdminClient patches the Dialer's SQL Admin API client with one that
+// uses the provided HTTP client and endpoint.
+func replaceSQLAdminService(d *Dialer, client *http.Client, endpoint string) {
+	svc, err := sqladmin.NewService(
+		context.Background(),
+		option.WithHTTPClient(client),
+		option.WithEndpoint(endpoint),
+	)
+	if err != nil {
+		panic(err)
```

Comment:
Panic's seem bad - even in tests? Is there a specific reason to use it here?

Reply:
For test setup functions, especially when the chance of failure is small, using panic avoids the boilerplate of checking for errors and then failing the test. We could ignore the error, but in the very small chance something has gone wrong, a panic stops the tests from resulting in hard to understand errors. For these kind of functions (e.g., generating an RSA key), we don't really care to know what the error is, just that it didn't happen.

Reply:
If you can bear with me for another PR or two, I'd like to show you how this pattern plays out. If you still don't like it on the last PR, I'll update with `t` pointers and an associated `Fatalf`.

Reply:
Ok